### ISSUE TITLE
Shopping List will also collect Tools

### ISSUE THREAD

**Issue by GrimNotepad** [b-morgan/Skillet-Classic#196]
I'd like to have the ability to collect the tools when it is picking up reagents for the Queue. Times when tools are stored in the bank, it will make sure they aren't left behind.

Experimented with the code and found that a ListString is provided when the SkillIndex is called, parsing that could get the list of the tools.

```
tsTools = BuildListString(GetTradeSkillTools(index))
if tsTools ~= nil then
   for word in string.gmatch(tsTools, '([^,]+)') do
      print(word)
   end
end
```

**Comment by b-morgan** [comment]
The example code you provided gives a list of the (localized) names of the tools required. The shopping list code uses itemIDs so the first problem is converting the (localized) names to itemIDs. There's no Blizzard function to do this so Skillet-Classic would have to build and maintain this list. 

A second problem is that there may be multiple items that satisfy the tool requirement (I believe this may not be the case for Classic Era but the code works on Cataclysm Classic as well).

A third problem is that the shopping list code to retrieve items is not always executed when the tradeskill is open, so the index of the recipe isn't available. Code would need to be added to capture the tools required when they are available so that they can be fetched at the appropriate time (and on the appropriate character).

While your request seems simple enough, the implementation is complex. I'll do some research but I'm not sure this is doable.

**Comment by b-morgan** [comment]
[Skillet-Classic-2.06-alpha3](https://legacy.curseforge.com/wow/addons/skillet-classic/files/6056277) has added some chat commands to manipulate a **per player** toolData list and code to fetch (if necessary) items in the toolData list from the bank. The list is saved between sessions so you only need to create it manually once on each character and should include all tools from all professions known by the player which might be stored in the bank.

The commands are:

- "/skillet tooladd item[,value]"
- "/skillet tooldel item"
- "/skillet toolclear"
- "/skillet tooldump"

Where 'item' can be the itemID of the tool or a link to the tool. The 'value' is stored but isn't currently used.
The easiest method is to type '/skillet tooladd ' and then shift-Click the tool in your bags. Same for 'tooldel'.

This "manual" method avoids the complex problems of localization and possible alternatives for tools.

The code to fetch (if necessary) items in the toolData list is controlled by the option Queue Tools.

**Comment by b-morgan** [comment]
I haven't tested this code at all the places where the shopping list is consulted. This may result in buying tools you have in the bank when you visit a vendor that sells the tool(s). It may add the tools to any AH shopping list generated by plugins like Auctionator.

Please provide feedbank on if this meets your needs (and if not, why not) and any "side-effects" you discover.

**Comment by b-morgan** [comment]
[Skillet-Classic-2.06-alpha4](https://legacy.curseforge.com/wow/addons/skillet-classic/files/6056862) will work a little better. At least the merchants who sell tools won't try and sell you more.

**Comment by GrimNotepad** [comment]
I used /skillet tooladd and the link, i see it in the tooldump but i don't seem to be getting the tool from the bank when i press retrieve
![image](https://github.com/user-attachments/assets/6dcda399-49eb-40f1-b305-bf922a1d6999)

tried with another item queued that did have some items in the bank like rough griding stone and they were retrieved from the bank , but again no blacksmith hammer

**Comment by b-morgan** [comment]
Did you set the option? Perhaps my instructions weren't clear enough, sorry about that.

![image](https://github.com/user-attachments/assets/103cbe44-b383-4b13-8e69-71de423ccaf7)

If so, you should see the tools in the shopping list when you open your bank:

![image](https://github.com/user-attachments/assets/39e7567d-9328-45f2-a12c-b6c0ed98a569)

The option name should probably be "Fetch Tools" but the option already existed from a failed attempt in 2022 so rather than create a new option that would require localization, I just went with the existing name.

**Comment by GrimNotepad** [comment]
You are right, I didn't choose that option in the settings. I was juggling getting to Org for Onyxia buff hahaha, didn't read thoroughly enough.

chose that option and i see the tools int he shopping list now!
